[BUG] setuptools._distutils not available on Debian 10 (setuptools v40.8.0)

**Description**
tests.integration.ssh.test_state.SSHStateTest.test_show_highstate (from pytest) fails due to 
```
from setuptools._distutils.version import LooseVersion as _LooseVersion\nModuleNotFoundError: No module named \'setuptools._distutils\'
```

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)
Testing framework for Debian 10.
```
[Dashboard](https://jenkins.saltproject.io/)
[Salt PR Debian 10 amd64 Py3 Pytest Builds](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/)
[master](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/)
[#2776](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/)
[Test Results](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/)
[Test Results](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/junit/)
[tests.integration.ssh.test_state](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/junit/tests.integration.ssh.test_state/)
[SSHStateTest](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/junit/tests.integration.ssh.test_state/SSHStateTest/)
test_show_highstate
```

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [X] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [X] onedir packaging
- [ ] used bootstrap to install


**Steps to Reproduce the behavior**
Run test above

**Expected behavior**
Test to pass

**Screenshots**

```
self = <tests.integration.ssh.test_state.SSHStateTest testMethod=test_show_highstate>

    def tearDown(self):
        """
        make sure to clean up any old ssh directories
        """
        salt_dir = self.run_function("config.get", ["thin_dir"], wipe=False)
>       self.assertIsInstance(salt_dir, (str,))
E       AssertionError: {'_error': 'Failed to return clean data', 'stderr': 'Traceback (most recent call last):\n  File "/var/tmp/.root_0e6fa3_salt/salt-call", line 26, in <module>\n    from salt.scripts import salt_call\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/__init__.py", line 146, in <module>\n    import salt._logging  # isort:skip\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/_logging/__init__.py", line 12, in <module>\n    from salt._logging.impl import (\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/_logging/impl.py", line 28, in <module>\n    from salt._logging.handlers import DeferredStreamHandler  # isort:skip\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/_logging/handlers.py", line 16, in <module>\n    from salt.utils.versions import warn_until_date\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/utils/versions.py", line 21, in <module>\n    from setuptools._distutils.version import LooseVersion as _LooseVersion\nModuleNotFoundError: No module named \'setuptools._distutils\'', 'stdout': '', 'retcode': 1} is not an instance of (<class 'str'>,)
```

**Versions Report**
relenv testing with master branch


</details>

**Additional context**
Wondering how the original PR https://github.com/saltstack/salt/pull/63309 with the changes for setuptools._distutils.version passed successfully, or was reported as successful.

--- a/changelog/63350.fixed
+++ b/changelog/63350.fixed
@@ -1,0 +1,1 @@
+Import StrictVersion and LooseVersion from setuptools.distutils.verison or setuptools._distutils.version, if first not available

--- a/salt/utils/pkg/win.py
+++ b/salt/utils/pkg/win.py
@@ -80,9 +80,13 @@ except ImportError:
             LooseVersion,  # pylint: disable=blacklisted-module
         )
     except ImportError:
-        from setuptools._distutils.version import (
-            LooseVersion,  # pylint: disable=blacklisted-module
-        )
+        try:
+            from setuptools._distutils.version import (
+                LooseVersion,  # pylint: disable=blacklisted-module
+            )
+        except ImportError:
+            # pylint: disable=blacklisted-module
+            from distutils.version import LooseVersion
 
 # pylint: disable=too-many-instance-attributes
 

--- a/salt/utils/pkg/win.py
+++ b/salt/utils/pkg/win.py
@@ -72,21 +72,7 @@ try:
 except ImportError:
     from collections import OrderedDict
 
-try:
-    from salt.utils.versions import LooseVersion
-except ImportError:
-    try:
-        from setuptools.distutils.version import (
-            LooseVersion,  # pylint: disable=blacklisted-module
-        )
-    except ImportError:
-        try:
-            from setuptools._distutils.version import (
-                LooseVersion,  # pylint: disable=blacklisted-module
-            )
-        except ImportError:
-            # pylint: disable=blacklisted-module
-            from distutils.version import LooseVersion
+from salt.utils.versions import LooseVersion
 
 # pylint: disable=too-many-instance-attributes
 

--- a/salt/utils/pkg/win.py
+++ b/salt/utils/pkg/win.py
@@ -75,9 +75,14 @@ except ImportError:
 try:
     from salt.utils.versions import LooseVersion
 except ImportError:
-    from setuptools._distutils.version import (
-        LooseVersion,  # pylint: disable=blacklisted-module
-    )
+    try:
+        from setuptools.distutils.version import (
+            LooseVersion,  # pylint: disable=blacklisted-module
+        )
+    except ImportError:
+        from setuptools._distutils.version import (
+            LooseVersion,  # pylint: disable=blacklisted-module
+        )
 
 # pylint: disable=too-many-instance-attributes
 

--- a/salt/utils/versions.py
+++ b/salt/utils/versions.py
@@ -18,16 +18,37 @@ import numbers
 import sys
 import warnings
 
-from setuptools._distutils.version import LooseVersion as _LooseVersion
-from setuptools._distutils.version import StrictVersion as _StrictVersion
+log = logging.getLogger(__name__)
+
+try:
+    from setuptools.distutils.version import LooseVersion as _LooseVersion
+except ImportError:
+    log.debug(
+        "unable to import LooseVersion from setuptools.distutils.version, will attempt setuptools._distutils.version"
+    )
+    try:
+        from setuptools._distutils.version import LooseVersion as _LooseVersion
+    except:
+        log.debug("unable to import LooseVersion from setuptools._distutils.version")
+        raise ImportError()
+
+try:
+    from setuptools.distutils.version import StrictVersion as _StrictVersion
+except ImportError:
+    log.debug(
+        "unable to import StrictVersion from setuptools.distutils.version, will attempt setuptools._distutils.version"
+    )
+    try:
+        from setuptools._distutils.version import StrictVersion as _StrictVersion
+    except:
+        log.debug("unable to import StrictVersion from setuptools._distutils.version")
+        raise ImportError()
+
 
 # pylint: enable=blacklisted-module
 import salt.version
 
 # pylint: disable=blacklisted-module
-
-
-log = logging.getLogger(__name__)
 
 
 class StrictVersion(_StrictVersion):

--- a/salt/utils/versions.py
+++ b/salt/utils/versions.py
@@ -22,26 +22,16 @@ log = logging.getLogger(__name__)
 
 try:
     from setuptools.distutils.version import LooseVersion as _LooseVersion
-except ImportError:
-    log.debug(
-        "unable to import LooseVersion from setuptools.distutils.version, will attempt setuptools._distutils.version"
-    )
-    try:
-        from setuptools._distutils.version import LooseVersion as _LooseVersion
-    except:
-        log.debug("unable to import LooseVersion from setuptools._distutils.version")
-        raise ImportError()
-
-try:
     from setuptools.distutils.version import StrictVersion as _StrictVersion
 except ImportError:
     log.debug(
-        "unable to import StrictVersion from setuptools.distutils.version, will attempt setuptools._distutils.version"
+        "unable to import from setuptools.distutils.version, will attempt setuptools._distutils.version"
     )
     try:
+        from setuptools._distutils.version import LooseVersion as _LooseVersion
         from setuptools._distutils.version import StrictVersion as _StrictVersion
     except:
-        log.debug("unable to import StrictVersion from setuptools._distutils.version")
+        log.debug("unable to import from setuptools._distutils.version")
         raise ImportError()
 
 

--- a/salt/utils/versions.py
+++ b/salt/utils/versions.py
@@ -18,33 +18,24 @@ import numbers
 import sys
 import warnings
 
-log = logging.getLogger(__name__)
-
+# pylint: disable=blacklisted-module
 try:
     from setuptools.distutils.version import LooseVersion as _LooseVersion
     from setuptools.distutils.version import StrictVersion as _StrictVersion
 except ImportError:
-    log.debug(
-        "unable to import from setuptools.distutils.version, will attempt setuptools._distutils.version"
-    )
     try:
         from setuptools._distutils.version import LooseVersion as _LooseVersion
         from setuptools._distutils.version import StrictVersion as _StrictVersion
     except ImportError:
-        log.debug("unable to import from setuptools._distutils.version")
-        try:
-            # pylint: disable=blacklisted-module
-            from distutils.version import LooseVersion as _LooseVersion
-            from distutils.version import StrictVersion as _StrictVersion
-        except ImportError:
-            log.debug("unable to import from distutils.version")
-            raise ImportError()
+        from distutils.version import LooseVersion as _LooseVersion
+        from distutils.version import StrictVersion as _StrictVersion
 
 
 # pylint: enable=blacklisted-module
 import salt.version
 
 # pylint: disable=blacklisted-module
+log = logging.getLogger(__name__)
 
 
 class StrictVersion(_StrictVersion):

--- a/salt/utils/versions.py
+++ b/salt/utils/versions.py
@@ -30,9 +30,15 @@ except ImportError:
     try:
         from setuptools._distutils.version import LooseVersion as _LooseVersion
         from setuptools._distutils.version import StrictVersion as _StrictVersion
-    except:
+    except ImportError:
         log.debug("unable to import from setuptools._distutils.version")
-        raise ImportError()
+        try:
+            # pylint: disable=blacklisted-module
+            from distutils.version import LooseVersion as _LooseVersion
+            from distutils.version import StrictVersion as _StrictVersion
+        except ImportError:
+            log.debug("unable to import from distutils.version")
+            raise ImportError()
 
 
 # pylint: enable=blacklisted-module

